### ISSUE TITLE
High cpu consumption

### ISSUE THREAD

**Issue by aacebedo** [denesb/xfce4-i3-workspaces-plugin#20]
I am facing a 100% CPU usage cause by the i3 workspace plugin. It does not happens everytime but reloading the i3 config several times helps to trigger the issue.
Do you have an idea?

**Comment by denesb** [comment]
By reloading the i3 config do you mean reloading (meta+Shift+c) or restart in-place (meta+Shift+r)?
The restart in-place is known not to work (#7), although it usually crashes instead of spinning the CPU.

**Comment by aacebedo** [comment]
Hi

Thanks for the answer. 
It happens both when I log out and log in and also, when I restart in place (more frequently) It does not happen when I reload the  config. 
Note that sometimes I also receive a message saying the plug-in failed to load when I use my vanilla fm xfce session. 
It looks like the plug-in is stuck searching for i3 workspace before they are really instantiated

**Comment by ivstog** [comment]
Hi,
I have the same issue. 
Along with the bug "Plugin unexpectedly left the panel" right at sturtup.

**Comment by Luke-Nukem** [comment]
Confirmed. There was an extra process spawned which caused the high usage (to the point of being sluggish on an i7 cpu). I can't for the life of me remember what it was, and I no-longer use xfce+i3 (custom environment now).

You can see what's happening by using 'top' in a console.

**Comment by denesb** [comment]
After some investigation it looks like this is tightly related to #7. I'm working on a fix for both.

**Comment by drgnomage** [comment]
Looks like its a problem with the library, if that helps.

> ps aux | grep "19232" 

> "josh      2871 96.1  0.2 212072 19232 ?        R    16:09   4:16 /usr/lib/xfce4/panel/wrapper-1.0 /usr/lib/xfce4/panel/plugins/libi3workspaces.so 2 14690370 i3-workspaces i3 Workspaces Plugin Workspaces plugin for managing i3 window manager workspaces"

Is there any more info I could provide on this?

**Comment by denesb** [comment]
Yes, it's definitely related to the library. I'm actually waiting for [this issue](https://github.com/acrisci/i3ipc-glib/issues/11) to be fixed, I suspect that this is one of the causes of this problem. Actually it is really hard to debug and investigate but it always happens when i3 is not available on the domain socket. This causes the library to crash or use 100% CPU (even after i3 becomes available).

**Comment by matjaz132** [comment]
Any new fixes for this issue?

**Comment by denesb** [comment]
I'm waiting for a new release of [i3ipc-glib](https://github.com/acrisci/i3ipc-glib)

**Comment by koopa1338** [comment]
I have 2 Monitors and therefor 2 i3 different workspaceplugins at startup. There are 2 wrapper processes that are at 80-100% load. It can be fixed bei sigterm these processes (they do restart anyway). This happens at every startup and the issue is that i cant see my workspacenumbers in my bars before I "restart" these wrapper processes. I hope a new version of i3ipc-glib will fix this problem.
